src/frontend/src/lib/utils/carousel.utils.ts

```diff
@@ -0,0 +1,96 @@
+import { isNullish } from '@dfinity/utils';
+
+interface CommonParams {
+	sliderFrame: HTMLElement | undefined;
+}
+
+/**
+ * A util for building carousel slider frame item.
+ */
+export const buildCarouselSliderFrameItem = ({
+	slide,
+	totalSlides
+}: {
+	slide: Node;
+	totalSlides: number;
+}): HTMLDivElement => {
+	// Create slider frame item and apply styling
+	const frameItem = document.createElement('div');
+	frameItem.style.width = `${100 / totalSlides}%`;
+
+	frameItem.appendChild(slide);
+
+	return frameItem;
+};
+
+/**
+ * A util for building carousel slider frame.
+ */
+export const buildCarouselSliderFrame = ({
+	slides,
+	slideWidth
+}: {
+	slides: Node[];
+	slideWidth: number;
+}): HTMLDivElement => {
+	// We add the last slide as the first element to properly animate last-to-first slide transition.
+	// The same applies to the last element - we use the first slide as the last item of the array.
+	const extendedSlides = [slides[slides.length - 1], ...slides, slides[0]];
+	const totalSlides = extendedSlides.length;
+
+	// Create slider frame and apply styling
+	const sliderFrame = document.createElement('div');
```

Comment:
Do we need to create this element? It can be an existing reference?

Reply:
With some adjustments - yes. That was a good idea 👍